ng/src/web/components/form/multiselect.js

```diff
@@ -0,0 +1,263 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+import 'core-js/fn/string/includes';
+
+import React from 'react';
+
+import glamorous from 'glamorous';
+
+import Downshift from 'downshift';
+
+import {arrays_equal, is_array, is_defined, is_empty} from 'gmp/utils.js';
+
+import Layout from '../layout/layout.js';
+
+import PropTypes from '../../utils/proptypes.js';
+
+import {
+  ArrowButton,
+  ArrowIcon,
+  Box,
+  case_insensitive_filter,
+  Input,
+  Item,
+  ItemContainer,
+  Menu,
+  option_items,
+  SelectContainer,
+  SelectedValue,
+} from './selectelements.js';
+
+const MultiSelectedValue = glamorous(SelectedValue)({
+  display: 'inline',
+  border: '1px solid #aaa',
+  borderRadius: '3px',
+  padding: '0 3px',
+  marginRight: '4px',
+  marginTop: '1px',
+  marginBottom: '1px',
+  backgroundColor: '#ddd',
+  fontFamily: 'Trebuchet MS, Tahoma, Verdana, Arial, sans-serif',
+  fontSize: '0.8em',
+});
+
+const DeleteButton = glamorous.div({
+  display: 'inline',
+  color: '#888',
+  marginRight: '2px',
+  '&:hover': {
+    color: '#000',
+  },
+});
+
+class MultiSelect extends React.Component {
+
+  constructor(...args) {
+    super(...args);
+
+    const {value} = this.props;
+
+    this.state = {
+      search: '',
+      selectedItems: is_array(value) ? value : [],
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleRemoveItem = this.handleRemoveItem.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleSelect = this.handleSelect.bind(this);
+    this.renderItem = this.renderItem.bind(this);
+  }
+
+  handleChange(value) {
+    const {name, onChange} = this.props;
+
+    if (is_defined(onChange)) {
+      onChange(value, name);
+    }
+  }
+
+  handleSearch(event) {
+    const {value} = event.target;
+    event.preventDefault(); // prevent input by downshift
+
+    this.setState({search: value});
+  }
+
+  handleSelect(selectedItem) {
+    const {selectedItems} = this.state;
+
+    if (selectedItems.includes(selectedItem)) {
+      this.handleRemoveItem(selectedItem);
+      return;
+    }
+
+    this.setState({
+      // add item
+      selectedItems: [...this.state.selectedItems, selectedItem],
+      // reset search term
+      search: '',
+    });
+  }
+
+  handleRemoveItem(item) {
+    const copy = [...this.state.selectedItems];
+    const index = copy.findIndex(elem => elem === item);
+    copy.splice(index, 1);
+    this.setState({
+      selectedItems: copy,
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {value} = nextProps;
+    if (arrays_equal(value, this.state.selectedItems)) {
+      this.setState({selectedItems: value});
+    }
+  }
+
+  renderItem(value, items, selectedItem) {
```

Comment:
selectedItem is unused and items is from this.props.items so calling renderItems can be simplified. see below